### ISSUE TITLE
`gauche.interactive#read-eval-print-loop` overhaul

### ISSUE THREAD

**Issue by shirok** [shirok/Gauche#1088]
Curently we have two `read-eval-print-loop`. 

`gauche#read-eval-print-loop` is the basic one.  Its basically like this, with some error handling etc.

```
(loop
  (prompter)
  (printer (evaluator (reader) env)))
```

`gauche.interactive#read-eval-print-loop` handles editable reader.   It is piggybacked on `gauche#read-eval-print-loop`, though, and there's some kludge.  Notably, with editable reader, prompt needs to be shown by the reader, isteadm of a separate prompter (the reader needs to know the columns the prompt occupies, and also redisplay it as needed.)

Without giving custom procedures, `gauche.interactive#read-eval-print-loop` creates its own internal reader and prompter that deals with it under the hood. 

However, that bandage breaks when the caller provides custom prompter to `gauche.interactive#read-eval-print-loop`.  That's the cause of https://github.com/shirok/Gauche/issues/1084.

Basically, repl with editable reader requires a different API, and pretending it to be the same as basic REPL
